### PR TITLE
Add Sonarlint VS Code extension

### PR DESCRIPTION
**Pull request by garagatyi**
### What does this PR do?
Add Sonarlint VS Code extension. It can showcase how to contribute VS Code extension to our registry.

### CODE REVIEW

**Comment by garagatyi** [comment]
@benoitf WDYT about this PR?

**Comment by benoitf** [comment]
@garagatyi I think it's nice. It's working on top of che.openshift.io ?

**Comment by garagatyi** [comment]
@benoitf Che on OSIO doesn't support it because it is not up to date yet. You are right, I'll need to postpone the merge

**Comment by dneary** [comment]
Very nice!

File: plugins/org.eclipse.che.vscode-sonarlint/0.0.1/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+id: org.eclipse.che.vscode-sonarlint
+version: 0.0.1
+type: VS Code extension
+name: vscode-sonarlint
+title: Sonarlint code intelligence
+description: VS Code extension that provides sonarlint features
+icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
+attributes:
+  extension: "vscode:extension/SonarSource.sonarlint-vscode"
+  container-image: "garagatyi/remotetheia:java"
```
Comment (l0rd) [inline plugins/org.eclipse.che.vscode-sonarlint/0.0.1/meta.yaml]: What about `eclipse/che-theia-plugin-base-java:latest` ?

Comment (garagatyi) [reply plugins/org.eclipse.che.vscode-sonarlint/0.0.1/meta.yaml]: Here is an issue to publish it https://github.com/eclipse/che/issues/12430

**Pull request merged**